studiopress/genesis-custom-blocks
# Add fields to the correct location, allow creating new block

Updated December 1st

#### Changes
* Can create a new block
* Can add a field to the 'Editor' or 'Inspector', depending on which is selected. 
* Can move fields within a location (like in the 'Editor' fields)
* Can move fields from one location to another (like from the 'Editor' fields to 'Inspector' fields)
* It was already possible to delete fields

#### Tickets
* [GF-2526](https://wpengine.atlassian.net/browse/GF-2526) (Add/remove fields)
* [GF-2528](https://wpengine.atlassian.net/browse/GF-2528) (Move/resize fields)
* [GF-2530](https://wpengine.atlassian.net/browse/GF-2530) (Inspector component)

#### Testing instructions
1. `composer install && npm install && npm run dev`
1. `/wp-admin` > Custom Blocks > Add New
1. Expected: something like this: <img width="1117" alt="bosf-sdf" src="https://user-images.githubusercontent.com/4063887/100181926-fa111280-2ea0-11eb-91ab-af14ff51a43c.png">
1. Please test deleting fields:
![the-filba](https://user-images.githubusercontent.com/4063887/100182299-e5814a00-2ea1-11eb-9ead-791ba90bee57.gif)
1. Create another new block, clicking `/wp-admin` > Custom Blocks > Add New
1. Please test moving fields. Moving isn't smooth like in Gutenberg, but it should work: ![creating-and-bo](https://user-images.githubusercontent.com/4063887/100785762-bce3dd80-33d6-11eb-9d10-7e89e7f702ba.gif)
1. Please test changing the location: 
![new-loc](https://user-images.githubusercontent.com/4063887/100786524-e6513900-33d7-11eb-9393-82ae1a1000af.gif)
1. Please test resizing: 
![chang-thidw](https://user-images.githubusercontent.com/4063887/100787684-a0957000-33d9-11eb-9d29-125f1efb21f5.gif)
